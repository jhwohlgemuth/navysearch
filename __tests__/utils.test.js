// import axios from 'axios';
// import moxios from 'moxios';
import {
    createMessageId,
    parseMessageUri,
    partitionByKeyLength
} from '../src/utils/index.js';

describe('Utilities', () => {
    test('can create message identifier strings', () => {
        const type = 'FOOBAR';
        const year = 20;
        const num = 42;
        expect(createMessageId({type, year, num})).toMatchSnapshot();
    });
    test('can parse full message URI', () => {
        const uri = 'https://www.public.navy.mil/bupers-npc/reference/messages/Documents/NAVADMINS/NAV2020/NAV20111.txt';
        const parsed = parseMessageUri(uri);
        expect(parsed).toMatchSnapshot();
    });
    test('can parse fragment of message URI', () => {
        const uri = '/bupers-npc/reference/messages/Documents/NAVADMINS/NAV2020/NAV20080.txt';
        const parsed = parseMessageUri(uri);
        expect(parsed).toMatchSnapshot();
    });
    test('can parse abnormal message URI', () => {
        const uri = 'https://www.public.navy.mil/bupers-npc/reference/messages/Documents/NAVADMINS/NAV2020/nav20092.txt';
        const parsed = parseMessageUri(uri);
        expect(parsed).toMatchSnapshot();
    });
    test('can split objects into smaller objects based on length of a certain key', () => {
        const MAX_LENGTH = 5;
        const a = {
            name: 'a',
            text: '123456789'
        };
        const b = {
            name: 'b',
            text: '12345'
        };
        const c = {
            name: 'c',
            text: '01234567890123456789'
        };
        const chunk = value => partitionByKeyLength('text', MAX_LENGTH, value);
        expect(chunk(a)).toMatchSnapshot();
        expect(chunk(b)).toMatchSnapshot();
        expect(chunk(c)).toMatchSnapshot();
        expect([a, b, c].flatMap(chunk)).toMatchSnapshot();
    });
});