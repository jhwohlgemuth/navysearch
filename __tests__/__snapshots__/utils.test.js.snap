// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities can create message identifier strings 1`] = `"FOOBAR2042"`;

exports[`Utilities can parse abnormal message URI 1`] = `
Object {
  "code": "NAV",
  "ext": ".txt",
  "id": "NAVADMIN20092",
  "num": "092",
  "type": "NAVADMIN",
  "url": "http://www.public.navy.milhttps://www.public.navy.mil/bupers-npc/reference/messages/Documents/NAVADMINS/NAV2020/nav20092.txt",
  "year": "20",
}
`;

exports[`Utilities can parse fragment of message URI 1`] = `
Object {
  "code": "NAV",
  "ext": ".txt",
  "id": "NAVADMIN20080",
  "num": "080",
  "type": "NAVADMIN",
  "url": "http://www.public.navy.mil/bupers-npc/reference/messages/Documents/NAVADMINS/NAV2020/NAV20080.txt",
  "year": "20",
}
`;

exports[`Utilities can parse full message URI 1`] = `
Object {
  "code": "NAV",
  "ext": ".txt",
  "id": "NAVADMIN20111",
  "num": "111",
  "type": "NAVADMIN",
  "url": "http://www.public.navy.milhttps://www.public.navy.mil/bupers-npc/reference/messages/Documents/NAVADMINS/NAV2020/NAV20111.txt",
  "year": "20",
}
`;

exports[`Utilities can split objects into smaller objects based on length of a certain key 1`] = `
Array [
  Object {
    "name": "a",
    "text": "12345",
  },
  Object {
    "name": "a",
    "text": "6789",
  },
]
`;

exports[`Utilities can split objects into smaller objects based on length of a certain key 2`] = `
Array [
  Object {
    "name": "b",
    "text": "12345",
  },
]
`;

exports[`Utilities can split objects into smaller objects based on length of a certain key 3`] = `
Array [
  Object {
    "name": "c",
    "text": "01234",
  },
  Object {
    "name": "c",
    "text": "56789",
  },
  Object {
    "name": "c",
    "text": "01234",
  },
  Object {
    "name": "c",
    "text": "56789",
  },
]
`;

exports[`Utilities can split objects into smaller objects based on length of a certain key 4`] = `
Array [
  Object {
    "name": "a",
    "text": "12345",
  },
  Object {
    "name": "a",
    "text": "6789",
  },
  Object {
    "name": "b",
    "text": "12345",
  },
  Object {
    "name": "c",
    "text": "01234",
  },
  Object {
    "name": "c",
    "text": "56789",
  },
  Object {
    "name": "c",
    "text": "01234",
  },
  Object {
    "name": "c",
    "text": "56789",
  },
]
`;
